server/src/main/java/org/opensearch/cluster/ClusterModule.java

```diff
@@ -474,4 +475,9 @@ public void setExistingShardsAllocators(GatewayAllocator gatewayAllocator, Shard
         allocationService.setExistingShardsAllocators(existingShardsAllocators);
     }
 
+    public void setRerouteServiceForAllocator(RerouteService rerouteService) {
+        if (shardsAllocator instanceof BalancedShardsAllocator) {
+            shardsAllocator.setRerouteService(rerouteService);
+        }
```

Comment:
we don't need the check here, the classes can themselves based on the overridden methods handle it?